image-customize

```diff
@@ -154,14 +153,19 @@ def main():
     machine.start()
     machine.wait_boot()
     try:
-        if args.upload:
-            upload_files(machine, args.upload)
-        if args.run_command:
-            run_command(machine, args.run_command)
-        if args.install:
-            install_packages(machine, args.install)
-        if args.script:
-            run_script(machine, args.script)
+        for (action, arg) in args.actions:
```

Comment:
Let's get rid of the if/elif ladder and have different subtypes of `ImageAction` instead, each of which has its own `invoke()` method that does the thing in question.

Reply:
subtypes don't work, as there is just one Action instance which mops up all the arguments. However, I did this differently now, let's see how you like it.

Reply:
Actually no, I guess it *might* instantiate the Action class multiple times.. But anyway, this is not obvious to wire up. Let's keep it simple.